store/rootmulti/store_test.go

```diff
@@ -672,6 +683,59 @@ func TestSetInitialVersion(t *testing.T) {
 	require.True(t, iavlStore.VersionExists(5))
 }
 
+func TestSetListenersAndListeningEnabled(t *testing.T) {
+	db := dbm.NewMemDB()
+	multi := newMultiStoreWithMounts(db, types.PruneNothing)
+	testKey := types.NewKVStoreKey("listening_test_key")
+	enabled := multi.ListeningEnabled(testKey)
+	require.False(t, enabled)
+
+	multi.SetListeners(testKey, []types.WriteListener{})
+	enabled = multi.ListeningEnabled(testKey)
+	require.False(t, enabled)
+
+	mockListener := types.NewStoreKVPairWriteListener(nil, nil)
+	multi.SetListeners(testKey, []types.WriteListener{mockListener})
```

Comment:
there's something which I didn't figure out how it was wired up by reading the PR. How is the listener passed down to the listenkv.Store when calling `SetListeners`? aka when is `listenkv.Store#listeners` updated?

I'd like to see the mockListener being a real listener:
```
mockListener := types.NewStoreKVPairWriteListener(new(bytes.Buffer), someMarshaler)
```

Then, after calling `SetListeners`, we call `.Set()` or `.Delete()` on the store, and we check the listener's output.

Reply:
Right now a listen.KVStore is only wrapped around a KVStore (with the appropriate `WriteListener`s) when they are returned by the multistore's `GetKVStore` [method](https://github.com/i-norden/cosmos-sdk/blob/adr038_writelistener/store/rootmulti/store.go#L501) (same approach as with the trace.KVStore wrapping). 

I'll expand the [TestGetListenWrappedKVStore ](https://github.com/i-norden/cosmos-sdk/blob/adr038_writelistener/store/rootmulti/store_test.go#L707) to call `Set` and `Delete` on the returned KVStore and demonstrate the listening is occurring on them!